### PR TITLE
Fix leaderboard polling logic

### PR DESCRIPTION
**Pull request by gautamjajoo**
There were two problems in the polling of the leaderboard.

1. There are multiple simultaneous requests while polling the leaderboard
2. When a user leaves the challenge component and visits it again the poller is started automatically (without going to leaderboard tab)

### Fixes

1. The cause of the issue was `selectPhaseSplit` was getting called twice, [1](https://github.com/Cloud-CV/EvalAI/blob/master/frontend_v2/src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.ts#L307) and [2](https://github.com/Cloud-CV/EvalAI/blob/master/frontend_v2/src/app/components/utility/selectphase/selectphase.component.html#L76). 
Fixed it by making routing changes for the child component and using `selectPhaseSplit` in the parent component.

2. Added a condition for the poller to start only if the user is at specific phase split's leaderboard.

https://user-images.githubusercontent.com/24366008/118995980-31223580-b9a5-11eb-97fb-b43f91a93da7.mp4

@Ram81 @Kajol-Kumari

### CODE REVIEW

File: frontend_v2/src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.html [hunk 1]
```diff
@@ -23,7 +23,7 @@ <h5 class="fw-light">Leaderboard</h5>
             [phases]="filteredPhaseSplits"
             [phaseSelectionType]="phaseSelectionType"
             [phaseSelectionListType]="phaseSelectionListType"
-            [phaseSplitSelected]="phaseSplitSelected()"
+            [phaseSelectedUrlChange]="phaseSelectedUrlChange"
```
Comment (Kajol-Kumari) [inline frontend_v2/src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.html:26]: Can we please rename it to `selectedPhaseSplitUrlChange`

File: frontend_v2/src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.ts [hunk 2]
```diff
@@ -345,7 +353,7 @@ export class ChallengeleaderboardComponent implements OnInit, AfterViewInit, OnD
         SELF.isSelectedPhaseLeaderboardPublic = selectedPhase.leaderboard_public
       }
     };
-  }
+
```
Comment (Kajol-Kumari) [inline frontend_v2/src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.ts:356]: please remove the extra spaces from here

File: frontend_v2/src/app/components/utility/selectphase/selectphase.component.ts [hunk 3]
```diff
@@ -22,9 +22,9 @@ export class SelectphaseComponent implements OnInit, OnChanges {
   @Input() phaseSelected: any;
 
   /**
-   * Selected phase split callback
+   * Selected phase split callback to update the router URL
    */
-  @Input() phaseSplitSelected: any;
+  @Input() phaseSelectedUrlChange: any;
```
Comment (Kajol-Kumari) [inline frontend_v2/src/app/components/utility/selectphase/selectphase.component.ts:27]: Same here. Can we please rename it to `selectedPhaseSplitUrlChange`

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Functionality wise, it looks good to me. Please do the minor requested enhancements.

**Comment by gautamjajoo** [comment]
@Kajol-Kumari Done the required changes!

**Review by Kajol-Kumari — APPROVED** [review]
LGTM 👍

**Review by Ram81 — APPROVED** [review]
LGTM 👍

**Pull request merged**